add time and duration to tasks

SpaceCowboy1 opened the pull request:
add validation of intersections
add tests

spaceCowboy11 committed:
add time and duration to tasks
add validation of intersections
add tests

avfyodorov commented on src/controllers/InMemoryTaskManager.java:
```diff
@@ -202,4 +250,7 @@ public List<Task> getHistory() {
         return historyManager.getHistory();
     }
 
+    public Set<Task> getPrioritizedTasks() {
```
Здесь нужно возвращать список, то есть List, не Set

avfyodorov commented on src/controllers/InMemoryTaskManager.java:
```diff
@@ -13,32 +14,41 @@ public class InMemoryTaskManager implements TaskManager {
     private HashMap<Integer, SubTask> subTasks = new HashMap<>();
     private HashMap<Integer, Task> tasks = new HashMap<>();
     private HistoryManager historyManager = Managers.getDefaultHistory();
+    Set<Task> prioritizedTasks = new TreeSet<>(Comparator.comparing(Task::getStartTime,
```
Для соблюдения инкапсуляции отсортированный список тоже лучше сделать приватным.

avfyodorov reviewed (commented):
Добрый день, Даниил! 
Очень хорошая работа, от меня всего пара уточнений. 

И ещё. Просьба добавить пару тестов на проверку отсортированного списка и продолжительности Эпика. Чтобы было видно, что работает Проверка пересечения по времени и правильно вычисляется продолжительность Эпика, у которого две-три подзадачи.

spaceCowboy11 committed:
изменен модификатор доступа prioritizedTasks
Добавлен тест на продолжительность эпика
Добавлен тест отсортированного списка prioritizedTasks

avfyodorov reviewed (commented):
Добрый день, Даниил!

Замечаний нет.
Работа принята.
